Sources/Utility/which.swift

```diff
@@ -10,10 +10,15 @@
 
 import Foundation
 
+private var memoized = [String: String]()
+
 public func which(_ arg0: String) -> String {
     if arg0.isAbsolute {
         return arg0
+    } else if let fullpath = memoized[arg0] {
+        return fullpath
     } else if let fullpath = try? Utility.popen(["/bin/sh", "-c", "which \(arg0)"]) {
+        memoized[arg0] = fullpath.chomp()
```

Comment:
I've added this in a new commit. It's a nice way as well to improve the test suite performance;

| Context                     | Linux | OS X  |
| --------------------------- | ----- | ----- |
| Baseline (master@827b703)   | 105 s | 141 s |
| Shell out to which          | 104 s | 142 s |
| lookup PATH                 | 110 s | 144 s |
| lookup PATH + memo          | 104 s | 140 s |